specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11-preview/searchservice.json

```diff
@@ -3956,6 +4152,329 @@
+			"name": {
+				"type": "string",
+				"description": "The name of the input."
+			},
+			"source": {
+				"type": "string",
+				"description": "The source of the input."
+			}
+		},
+		"required": ["name",
+		"source"],
+		"description": "Input field mapping for a skill."
+	},
+	"OutputFieldMappingEntry": {
+		"properties": {
+			"name": {
+				"type": "string",
+				"description": "The name of the output."
+			},
+			"targetName": {
+				"type": "string",
+				"description": "The target name of the output."
+			}
+		},
+		"required": ["name",
+		"targetName"],
+		"description": "output field mapping for a skill."
+	},
+	"KeyPhraseExtractionSkill": {
+		"x-ms-discriminator-value": "#Microsoft.Skills.Text.KeyPhraseExtractionSkill",
+		"allOf": [{
+			"$ref": "#/definitions/Skill"
+		}],
+		"properties": {
+			"defaultLanguageCode": {
+				"type": "string",
+				"default": "en",
+				"description": "A value indicating which language code. Default is en."
+			},
+			"maxKeyPhraseCount": {
+				"type": "integer",
+				"format": "int32",
+				"x-nullable": true
+			}
+		},
+		"description": "Text analytics key phrase extraction."
+	},
+	"OcrSkill": {
+		"x-ms-discriminator-value": "#Microsoft.Skills.Vision.OcrSkill",
+		"allOf": [{
+			"$ref": "#/definitions/Skill"
+		}],
+		"properties": {
+			"textExtractionAlgorithm": {
+				"type": "string",
+				"default": "printed",
+				"description": "A value indicating which algorithm to use. Default is printed."
+			},
+			"defaultLanguageCode": {
+				"type": "string",
+				"default": "en",
+				"description": "A value indicating which language code. Default is en."
+			},
+			"detectOrientation": {
+				"type": "boolean",
+				"default": false,
+				"description": "A value indicating to turn orientation detection on or not. Default is false."
+			}
+		},
+		"description": "Computer vision OCR."
+	},
+	"ImageAnalysisSkill": {
+		"x-ms-discriminator-value": "#Microsoft.Skills.Vision.ImageAnalysisSkill",
+		"allOf": [{
+			"$ref": "#/definitions/Skill"
+		}],
+		"properties": {
+			"defaultLanguageCode": {
+				"type": "string",
+				"default": "en",
+				"description": "A value indicating which language code. Default is en."
+			},
+			"visualFeatures": {
+				"type": "array",
+				"items": {
+					"type": "string"
+				},
+				"description": "A list of visual features."
+			},
+			"details": {
+				"type": "array",
+				"items": {
+					"type": "string"
+				},
+				"description": "Celebrities or Landmarks."
+			}
+		},
+		"description": "Computer vision image analyzing."
+	},
+	"LanguageDetectionSkill": {
+		"x-ms-discriminator-value": "#Microsoft.Skills.Text.LanguageDetectionSkill",
+		"allOf": [{
+			"$ref": "#/definitions/Skill"
+		}],
+		"description": "The skill for detecting language."
+	},
+	"ShaperSkill": {
+		"x-ms-discriminator-value": "#Microsoft.Skills.Util.ShaperSkill",
+		"allOf": [{
+			"$ref": "#/definitions/Skill"
+		}],
+		"description": "The skill for reshaping the outputs."
```

Comment:
No other properties for this skill?

Reply:
No. There is not. I have added more details in the description. Thanks.